sql/plan/cross_join.go

```diff
@@ -155,17 +155,18 @@ func (i *crossJoinIterator) Next() (sql.Row, error) {
 	}
 }
 
-func (i *crossJoinIterator) Close() error {
-	if err := i.l.Close(); err != nil {
-		if i.r != nil {
-			_ = i.r.Close()
-		}
-		return err
+func (i *crossJoinIterator) Close() (err error) {
+	if i.l != nil {
+		err = i.l.Close()
 	}
 
 	if i.r != nil {
-		return i.r.Close()
+		if err != nil {
+			err = i.r.Close()
```

Comment:
So, if there is an error closing `i.l` but no error closing `i.r` `err` will be nil and the first error will be "forgotten", is that ok?

Reply:
good catch, thanks.
Other way around - Fix already sent.